pkg/gpu/probe.go

```diff
@@ -48,12 +49,13 @@ type ProbeDependencies struct {
 
 // Probe represents the GPU monitoring probe
 type Probe struct {
-	mgr      *ddebpf.Manager
-	cfg      *Config
-	consumer *cudaEventConsumer
-	attacher *uprobes.UprobeAttacher
-	deps     ProbeDependencies
-	sysCtx   *systemContext
+	mgr       *ddebpf.Manager
+	cfg       *Config
+	consumer  *cudaEventConsumer
+	attacher  *uprobes.UprobeAttacher
+	generator *statsGenerator
+	deps      ProbeDependencies
+	sysCtx    *systemContext
```

Comment:
i don't like the parameter propagation pattern. it surfaces design flaws, as the code is getting more complicated to follow and understand.

example: ebpf_telemetry that we had in the past were we would propagate that pointer everywhere in the codebase.

in this example i do understand the rational, but it has to be a good logical explanation why would we need this context propagated into multiple area. I still suggest for simplicity removing this field from probe for now, and in case more use-cases will arise, we can refactor it

Reply:
Removed from the probe then. In any case I think we need to retain that system context, the same information is going to be required by multiple parts of the code and if we don't want to pass it around, either we go with a global variable (which I don't think is appropriate) or with some kind of dependency-injection which I feel is going to be more complex for such a small thing.